CODING_GUIDELINES.md

```diff
@@ -151,6 +151,66 @@ prior to attempting a search.
 Note: We are in the process of replacing methods that don't adhere to the above principles (for example, return a
 structure instead of a pointer, return a nil error on not-found, etc).
 
+## Implementing functions to support different vCD API versions
+
+Functions dealing with different versions should use a matrix structure to identify which calls to run according to the 
+highest API version supported by vCD. An example can be found in adminvdc.go.
+
+Note: use this pattern for adding new vCD functionality, which is not available in the earliest API version supported 
+by the code base (as indicated by `Client.APIVersion`).
+
+```
+type vdcVersionedFunc struct {
+	SupportedVersion string
+	CreateVdc        func(adminOrg *AdminOrg, vdcConfiguration *types.VdcConfiguration) (*Vdc, error)
+	CreateVdcAsync   func(adminOrg *AdminOrg, vdcConfiguration *types.VdcConfiguration) (Task, error)
+	UpdateVdc        func(adminVdc *AdminVdc) (*AdminVdc, error)
+	UpdateVdcAsync   func(adminVdc *AdminVdc) (Task, error)
+}
+
+var vdcVersionedFuncsV90 = vdcVersionedFuncs{
+	SupportedVersion: "29.0",
+	CreateVdc:        createVdc,
+	CreateVdcAsync:   createVdcAsync,
+	UpdateVdc:        updateVdc,
+	UpdateVdcAsync:   updateVdcAsync,
+}
+
+var vdcVersionedFuncsV97 = vdcVersionedFuncs{
+	SupportedVersion: "32.0",
+	CreateVdc:        createVdcV97,
+	CreateVdcAsync:   createVdcAsyncV97,
+	UpdateVdc:        updateVdcV97,
+	UpdateVdcAsync:   updateVdcAsyncV97,
+}
+
+var vdcVersionedFuncsByVcdVersion = map[string]vdcVersionedFuncs{
+	"vdc9.0":  vdcCrudV90,
```

Comment:
Please note that `vcdCrudV90` is not defined.

Reply:
fixed